### ISSUE TITLE
QOS statistics from Oracle SBC via Horaclifix

### ISSUE THREAD

**Issue by signalAU** [sipcapture/homer#263]
Hi Team,

Im new to linux and all things Git and need some help getting QOS into Homer.

I have a working Homer 5.0.5 install using Horaclifix (GO) to retrieve data from a pair of SBC's.

I can successfully see calls in Homer and have updated my install with the details from issue #248 (MosinAnton)

I have confirmed I am receiving IPFIX data type = SetID 268 (qos data from SBC) however my DB tables are not populating with any RTCP data. I have enabled RTCP partitions in my Kamailio.cfg file also and daily tables are being created successfully for RTCP.

Is there a way that I can check for resultant HEP packets from Horaclifix to Homer to see if QOS data is there?

I have used the files for Horaclifix 0.8.0 to update my kamailio.cfg and DB rotation scripts however I am still not getting any RTCP data in any table in my DB.

Should I have patched my install a different way instead of how it is suggest in issue #248?

Any assistance greatly appreciated.

Thanks.

**Comment by signalAU** [comment]
**UPDATE**

I reverted to an older Kamailio.cfg file whilst maintaining the remaining updated files in the Horaclifix 0.8.0 zip and I can now finally see data in my daily rtcp_capture_all_YYYYMMDD tables which is great.

Furthermore, I had to enable 'qos-enable' under the SBC realm-config to receive realm-specific QOS data.

If i turn debug on and try to run the SQL queries from the UI manually, I get 0 results.

However If I select * from the rtcp tables I gets lots of data there.

Could there be some issue with correlation IDs?

Thanks.

**Comment by signalAU** [comment]
Correction, I can see details in the report_capture_all_YYYYMMDD tables but not the rtcp_capture_all_YYYYMMDD tables.... I got a little excited there....


|  1 | 2018-01-22 18:44:45 | 1516607995000000 | b0d33dc116a05d72f53a68c6d1d27d7ee754c1c5@XXX.XXX.2.131 | 0.0.0.0        |           0 | XXX.XXX.223.130 |            44686 |    17 |      3 |   38 | horaclifix | {"CALLEE_DST_IP":"10.0.16.10","CALLEE_DST_PORT":28116,"CALLEE_SRC_IP":"XXX.XXX.223.130","CALLEE_SRC_PORT":44686,"CALLEE_VLAN":0,"CALLER_DST_IP":"172.18.1.200","CALLER_DST_PORT":19162,"CALLER_SRC_IP":"0.0.0.0","CALLER_SRC_PORT":0,"CALLER_VLAN":0,"INC_ID":"b0d33dc116a05d72f53a68c6d1d27d7ee754c1c5@XXX.XXX.2.131","INC_MOS":439,"INC_REALM":"TCC-PSTN","INC_RTCP_AVG_JITTER":5,"INC_RTCP_AVG_LAT":26,"INC_RTCP_BYTE":3380,"INC_RTCP_MAX_JITTER":7,"INC_RTCP_MAX_LAT":27,"INC_RTCP_PK":26,"INC_RTCP_PK_LOSS":0,"INC_RTP_AVG_JITTER":0,"INC_RTP_BYTE":263562,"INC_RTP_MAX_JITTER":1,"INC_RTP_PK":1209,"INC_RTP_PK_LOSS":0,"INC_RVAL":9244,"MEDIA_TYPE":1,"OUT_ID":"b0d33dc116a05d72f53a68c6d1d27d7ee754c1c5@XXX.XXX.2.131","OUT_MOS":438,"OUT_REALM":"TCC-INTERNAL","OUT_RTCP_AVG_JITTER":0,"OUT_RTCP_AVG_LAT":47,"OUT_RTCP_BYTE":1010,"OUT_RTCP_MAX_JITTER":0,"OUT_RTCP_MAX_LAT":48,"OUT_RTCP_PK":7,"OUT_RTCP_PK_LOSS":0,"OUT_RTP_AVG_JITTER":0,"OUT_RTP_BYTE":261818,"OUT_RTP_MAX_JITTER":4,"OUT_RTP_PK":1201,"OUT_RTP_PK_LOSS":0,"OUT_RVAL":9217}          |
|  2 | 2018-01-22 18:48:41 | 1516607555000000 | 8c977680-a651972a-c1e66-40212ac@172.18.2.4             | 0.0.0.0        |           0 | 172.18.1.200    |            19148 |    17 |      3 |   38 | horaclifix | {"CALLEE_DST_IP":"XXX.XXX.12.182","CALLEE_DST_PORT":40138,"CALLEE_SRC_IP":"172.18.1.200","CALLEE_SRC_PORT":19148,"CALLEE_VLAN":0,"CALLER_DST_IP":"XXX.XXX.223.130","CALLER_DST_PORT":44720,"CALLER_SRC_IP":"0.0.0.0","CALLER_SRC_PORT":0,"CALLER_VLAN":0,"INC_ID":"8c977680-a651972a-c1e66-40212ac@172.18.2.4","INC_MOS":438,"INC_REALM":"TCC-INTERNAL","INC_RTCP_AVG_JITTER":0,"INC_RTCP_AVG_LAT":44,"INC_RTCP_BYTE":480,"INC_RTCP_MAX_JITTER":0,"INC_RTCP_MAX_LAT":45,"INC_RTCP_PK":4,"INC_RTCP_PK_LOSS":0,"INC_RTP_AVG_JITTER":0,"INC_RTP_BYTE":245686,"INC_RTP_MAX_JITTER":1,"INC_RTP_PK":1127,"INC_RTP_PK_LOSS":0,"INC_RVAL":9220,"MEDIA_TYPE":1,"OUT_ID":"8c977680-a651972a-c1e66-40212ac@172.18.2.4","OUT_MOS":439,"OUT_REALM":"TCC-PSTN","OUT_RTCP_AVG_JITTER":1,"OUT_RTCP_AVG_LAT":26,"OUT_RTCP_BYTE":2600,"OUT_RTCP_MAX_JITTER":2,"OUT_RTCP_MAX_LAT":27,"OUT_RTCP_PK":20,"OUT_RTCP_PK_LOSS":0,"OUT_RTP_AVG_JITTER":0,"OUT_RTP_BYTE":464340,"OUT_RTP_MAX_JITTER":1,"OUT_RTP_PK":2130,"OUT_RTP_PK_LOSS":0,"OUT_RVAL":9262}                               |
|  3 | 2018-01-22 18:56:57 | 1516608202000000 | 9e20edc11611898113a69614cbf6084bad64bfa@XXX.XXX.7.131  | XXX.XXX.11.216 |       40032 | XXX.XXX.221.130 |            45658 |    17 |      3 |   38 | horaclifix | {"CALLEE_DST_IP":"10.1.32.35","CALLEE_DST_PORT":18188,"CALLEE_SRC_IP":"XXX.XXX.221.130","CALLEE_SRC_PORT":45658,"CALLEE_VLAN":0,"CALLER_DST_IP":"172.18.2.200","CALLER_DST_PORT":20152,"CALLER_SRC_IP":"XXX.XXX.11.216","CALLER_SRC_PORT":40032,"CALLER_VLAN":0,"INC_ID":"9e20edc11611898113a69614cbf6084bad64bfa@XXX.XXX.7.131","INC_MOS":440,"INC_REALM":"TCC-PSTN","INC_RTCP_AVG_JITTER":1,"INC_RTCP_AVG_LAT":0,"INC_RTCP_BYTE":1170,"INC_RTCP_MAX_JITTER":4,"INC_RTCP_MAX_LAT":0,"INC_RTCP_PK":9,"INC_RTCP_PK_LOSS":0,"INC_RTP_AVG_JITTER":0,"INC_RTP_BYTE":465648,"INC_RTP_MAX_JITTER":5,"INC_RTP_PK":2136,"INC_RTP_PK_LOSS":0,"INC_RVAL":9284,"MEDIA_TYPE":1,"OUT_ID":"9e20edc11611898113a69614cbf6084bad64bfa@XXX.XXX.7.131","OUT_MOS":440,"OUT_REALM":"TCC-INTERNAL","OUT_RTCP_AVG_JITTER":0,"OUT_RTCP_AVG_LAT":0,"OUT_RTCP_BYTE":0,"OUT_RTCP_MAX_JITTER":0,"OUT_RTCP_MAX_LAT":0,"OUT_RTCP_PK":0,"OUT_RTCP_PK_LOSS":0,"OUT_RTP_AVG_JITTER":0,"OUT_RTP_BYTE":468046,"OUT_RTP_MAX_JITTER":2,"OUT_RTP_PK":2147,"OUT_RTP_PK_LOSS":0,"OUT_RVAL":9284}         |

**Comment by signalAU** [comment]
Ok, so what I have found and I probably misread this during setup, for Horaclifix QOS stats, the table used is 'report_capture' or 'report_capture_all_YYYYMMDD'.

Whereas other RTCP types use the 'rtcp_capture' or 'rtcp_capture_call_YYYYMMDD' tables.

However to use daily tables for Horaclifix QOS statistics - 'report_capture_all_YYYYMMDD' - I had to update my preferences.php with the following:

/* Partition rtcp_capture table. Set to 1 if you use rtcp_capture_all_YYYYMMDDD */
define('RTCP_TABLE_PARTITION', 1);
/* Partition report_capture table as above */
define('REPORT_TABLE_PARTITION', 1);

I now have working Horaclifix QOS stats in my Homer and displaying correctly...

I'll close this as resolved.
